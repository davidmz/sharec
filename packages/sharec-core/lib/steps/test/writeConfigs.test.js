const { vol } = require('memfs')
const writeConfigs = require('../writeConfigs')

describe('steps > writeConfigs', () => {
  const semaphore = {
    start: jest.fn(),
    success: jest.fn(),
    error: jest.fn(),
    fail: jest.fn(),
  }
  let context

  beforeEach(() => {
    jest.clearAllMocks()
    vol.reset()
  })

  it('writes merged configs', async () => {
    context = {
      targetPath: '/target',
      mergedConfigs: {
        '/target/foo.txt': 'foo',
        '/target/bar.txt': 'bar',
        '/target/baz.txt': 'baz',
      },
    }

    vol.fromJSON({}, '/')

    await writeConfigs(context, semaphore)

    expect(vol.readFileSync('/target/foo.txt', 'utf8')).toMatchSnapshot()
    expect(vol.readFileSync('/target/bar.txt', 'utf8')).toMatchSnapshot()
    expect(vol.readFileSync('/target/baz.txt', 'utf8')).toMatchSnapshot()
  })
})
