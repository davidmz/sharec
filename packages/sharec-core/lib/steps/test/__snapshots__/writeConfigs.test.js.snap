// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`steps > writeConfigs writes merged configs 1`] = `
"foo
"
`;

exports[`steps > writeConfigs writes merged configs 2`] = `
"bar
"
`;

exports[`steps > writeConfigs writes merged configs 3`] = `
"baz
"
`;
